s3_file_field/_multipart.py

```diff
@@ -67,8 +68,19 @@ def initialize_upload(
         ]
         return InitializedUpload(object_key=object_key, upload_id=upload_id, parts=parts)
 
-    def finalize_upload(self, finalization: UploadFinalization) -> None:
-        raise NotImplementedError
+    def finalize_upload(self, finalization: UploadFinalization) -> FinalizedUpload:
+        finalize_url = self._generate_presigned_finalize_url(finalization)
+        body = self.marshal_finalize_body(finalization)
+        return FinalizedUpload(finalize_url=finalize_url, body=body)
+
+    def marshal_finalize_body(self, finalization: UploadFinalization) -> str:
+        """
+        Generate the body of a presigned finalize request.
+
+        We use the internal minio implementation to generate the XML body of the request
```

Comment:
In the current library design, it's possible that `minio` is not installed at all. Is there a way to also get this body from `boto`?

Reply:
There is not, so I just did the serialization with f strings instead.